index.js

```diff
@@ -247,6 +248,12 @@ const plugin = fp(async function (app, opts) {
     }
 
     fastifyGraphQl.schema = extendSchema(fastifyGraphQl.schema, s)
+
+    const context = assignApplicationHooksToContext({}, fastifyGraphQl[kHooks])
+
+    if (context.onExtendSchema !== null) {
+      return onExtendSchemaHandler({ schema: fastifyGraphQl.schema, context })
```

Comment:
I don't think this implementation is technically breaking, because a promise is only returned if a hook is specified.